Update MASTG-TECH-0017, MASTG-TECH-0023 and MASTG-TOOL-0018 (#3346)
OWASP/owasp-mstg: tools/android/MASTG-TOOL-0018.md

source: https://github.com/skylot/jadx
---

> Jadx is an opensource Dex and Java decompiler available both in a CLI (`jadx`) and GUI (`jadx-gui`) form. It's actively developed and offers an extensive range of features that can help you when analyzing Android applications:

Reviewer: ```suggestion
Jadx is an open-source Dex and Java decompiler available both in a CLI (`jadx`) and GUI (`jadx-gui`) form. It's actively developed and offers an extensive range of features to help you analyze Android applications:
```

---

jadx (Dex to Java Decompiler) is a command line and [GUI tool](https://github.com/skylot/jadx/wiki/jadx-gui-features-overview "jadx gui features overview") for producing Java source code from Android DEX and APK files - <https://github.com/skylot/jadx>

> ## Decompiling via CLI

Reviewer: This looks like it belongs in the TECH

The @MASTG-APP-0003 app can be decompiled using `jadx` by specifying the output directory and the target APK:

```console

> jadx -d decompiled UnCrackable-Level1.apk

Reviewer: Add the single class option since we use it as well in https://github.com/cpholguera/MASTestApp-Android/blob/main/tools/extract-code-for-mastg-demo.sh#L35
